### ISSUE TITLE
[cleanup] Math/rand.Seed deprecated

### ISSUE THREAD

**Issue by n4x2** [gopasspw/gopass#2650]
### Summary
Since Go 1.20 release, math/rand.Seed is marked "Deprecated"

Looks like five instances to consider updating:
```sh
find . -name "*.go" -type f -print | xargs grep .Seed
```
```sh
./pkg/pwgen/pwgen_test.go:      mrand.Seed(1789)
./pkg/pwgen/rand.go:    rand.Seed(time.Now().Unix() + int64(os.Getpid()+os.Getppid()))
./pkg/fsutil/fsutil.go: rand.Seed(time.Now().UnixNano())
./pkg/gitconfig/config_test.go: rand.Seed(time.Now().Unix())
./internal/action/binary_test.go:       rand.Seed(42)
```

**Comment by dominikschulz** [comment]
Usually "Deprecated" just means it's finally stable ;)

But I'll check what the recommendation is these days.

**Comment by dominikschulz** [comment]
* `pkg/pwgen/rand.go` This one can Go away once we don't need to support Go 1.20 anymore.
* `pkg/fsutil/fsutil.go` Same here.

For the test files we actually rely on detemernistic output, so we will need some more changes. We need to introduce a global variable with a local generator and seed it like this `New(NewSource(seed))`. See https://pkg.go.dev/math/rand#Seed

**Comment by orangekame3** [comment]
@dominikschulz 
Hi, I'd like to take on this issue if that's okay.

**Comment by dominikschulz** [comment]
@orangekame3 Feel free to do so

**Comment by orangekame3** [comment]
@dominikschulz 
Thank you. I've created a PR. Please check it out:
https://github.com/gopasspw/gopass/pull/2675

**Comment by lhardt** [comment]
Hello @dominikschulz ! 

I see the last pull request on this issue is from last year, but I see the  `good-first-issue` tag. 
Do you mind if I open my own PR?

**Comment by AnomalRoil** [comment]
@lhardt Please go ahead. It should be as simple as taking the previous PR and fixing the merge conflicts and then doing the changes that were requested in there if you want. 
Or you can try and do it your way too if you have another idea to do it.

**Comment by lhardt** [comment]
@dominikschulz  Can this be marked as closed following #2953?
I guess #2873  can be closed following #2954 too

**Comment by dominikschulz** [comment]
Fixed by #2953
